flang/test/Lower/character-assignment.f90

```diff
@@ -1,44 +1,47 @@
-! RUN: bbc %s -o "-" -emit-fir | FileCheck %s
+! RUN: bbc %s -o - -emit-fir | FileCheck %s
+! RUN: bbc %s -o - | FileCheck --check-prefix=UNBOX %s
 
 ! Simple character assignment tests
+! UNBOX-LABEL: assign1
 ! CHECK-LABEL: assign1
 subroutine assign1(lhs, rhs)
   character(*, 1) :: lhs, rhs
   lhs = rhs
-  ! Unboxing
-  ! CHECK-DAG:[[lhs:%[0-9]+]]:2 = fir.unboxchar %arg0
-  ! CHECK-DAG:[[rhs:%[0-9]+]]:2 = fir.unboxchar %arg1
-
   ! Compute minimum length
-  ! CHECK-DAG:%[[cmp_len:[0-9]+]] = cmpi "slt", [[lhs]]#1, [[rhs]]#1
-  ! CHECK-DAG:[[min_len:%[0-9]+]] = select %[[cmp_len]], [[lhs]]#1, [[rhs]]#1
+  ! UNBOX-DAG: %[[lhs:.*]]:2 = fir.unboxchar %arg0
+  ! UNBOX-DAG: %[[rhs:.*]]:2 = fir.unboxchar %arg1
+  ! UNBOX: %[[cmp_len:[0-9]+]] = cmpi "slt", %[[lhs]]#1, %[[rhs]]#1
```

Comment:
Sorry, this was a global replace sort of action. I was getting strange mismatches and it looked like some of the ssa-values were getting names instead of numbers for some reason. I undoubtedly changed many that didn't really need to be. (Debugging these CHECK lines can be a real time sink.)

Reply:
OK, your change actually makes the tests more readable IMO, thanks.